### DISCUSSION TITLE
glfw/opengl on linux wayland - main window has no title bar and cannot be moved or resized properly

### DISCUSSION THREAD

**Discussion by lvraamv** [ocornut/imgui#7645] [New Users Build/Link/Run issues ONLY!]
Hello, 
I'm running Pop!_OS 22.04 with wayland and I managed to build the example provided in the docking branch with the glfw + opengl backend without issue. I also disabled multi view ports since it is not currently supported on wayland.

When running the application though I'm facing some issues:
- main window doesn't have a title bar (and no close button) and cannot be moved by dragging,
![Screenshot from 2024-06-02 00-27-35](https://github.com/ocornut/imgui/assets/68163404/7e0e4c81-d9fb-4f56-8a71-34e576107948)

- clicking anywhere on the window border (the white border) resizes the whole window gradually,
- dragging one of the window's side to resize it, resizes another direction (for example, dragging the top border to increase or decrease the window's height, increases its width at the same time, and vice-versa),
- the only way to actually resize the window is by dragging its corner and changing the height and width simultaneously,
- all the other features of the demo window seem to work correctly.

I installed glfw from my package manager: libglfw3, libglfw3-dev, libglfw3-wayland.

I tried going through the issues and discussions but couldn't find anything related to that. I'm getting used to linux and C++ development in general so I'd appreciate any help I can get.

Thanks

**Comment by inobelar** [comment]
@lvraamv Hello! As you can see, the contents of the window are rendered **correctly**, so first of all this is **not** a problem Dear ImGui.

Since you mentioned `Pop OS`, `wayland` and `glfw` - the problem is on their side:

- As far as I can see, `Pop OS` 22.04 uses GNOME Desktop Environment, so if you don't see problems with other applications (I assuming default uses `gtk` for window creation & graphics), the problem is most likely in the `glfw` <--> `wayland` interop.
- More generally, this resembles a problem with the [`libdecor`](https://gitlab.freedesktop.org/libdecor/libdecor) library, or its absence. It is she who is responsible for window decoration (title, border, minimze/close buttons) in the `wayland` environment. Try digging in this direction.

Also 'Dear ImGui', OpenGL, window creation, input handling, event triggering, display via wayland are working for you. `glfw` as a whole should also work ok, the problem is outside its plane, or `glfw` is built with the wrong configuration. 

This issue definitely does not concern 'Dear ImGui', better ask about it in [`glfw` issues](https://github.com/glfw/glfw/issues). Or even simply - try to build `glfw` [from sources](https://www.glfw.org/docs/3.3/compile.html) and next run [examples](https://github.com/glfw/glfw/tree/master/examples) and [test](https://github.com/glfw/glfw/tree/master/tests) to see how them look.

**Reply by lvraamv** [reply]
@inobelar thank you for the feedback as it helped me find a solution.

The issue seemed to be related to the installed "libglfw3-wayland" package. Since it uses the wayland backend, I assumed it was the package that I needed but, in fact, I had to install the default "libglfw3" package, that uses the x11 backend, instead.

Everything works as expected now. The window has a border and can be moved/resized correctly.
Thanks again!

**Reply by inobelar** [reply]
@lvraamv Great to see you resolved this! 👍 

Please [mark this discussion as "resolved"](https://docs.github.com/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer) ✅ to reduce the burden on 'Dear ImGui' maintainers.
